pandas/tseries/holiday.py

```diff
@@ -257,6 +264,11 @@ class from pandas.tseries.offsets, default None
         self.observance = observance
         assert days_of_week is None or type(days_of_week) == tuple
         self.days_of_week = days_of_week
+        self.exclude_dates = (
+            [Timestamp(ex) for ex in exclude_dates]
+            if exclude_dates is not None
+            else exclude_dates
+        )
```

Comment:
With the constructor argument being stricter, we can just validate the arguments are all `Timestamp`